precompiles/crowdloan-rewards/src/lib.rs

```diff
@@ -160,27 +124,30 @@ where
 			total, claimed
 		);
 
-		Ok(succeed(
-			EvmDataWriter::new().write(total).write(claimed).build(),
-		))
+		Ok((total, claimed))
 	}
 
-	fn claim(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+	#[precompile::public("claim()")]
+	fn claim(handle: &mut impl PrecompileHandle) -> EvmResult {
 		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
 		let call = pallet_crowdloan_rewards::Call::<Runtime>::claim {};
 
 		RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
 
-		Ok(succeed([]))
+		Ok(())
 	}
 
-	fn update_reward_address(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+	#[precompile::public("updateRewardAddress(address)")]
+	#[precompile::public("update_reward_address(address)")]
```

Comment:
I would be explicit in the fac that this is non-payable. I guess that this si the default mode but I think it helps visualizing

Reply:
Well it is non-payable and non-static(non-view). I used the same default as in Solidity, and prevent at build time to both add `payable` and `view` :)